docs/en/platform-engineers/cloud-services.md

```diff
@@ -58,86 +65,121 @@ EOF
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: Application
+metadata:
+  name: mydatabase
+spec:
+  components:
+    - name: myrds
+      type: rds
+      properties:
+        name: "alibaba-rds"
+        storage: 20
+        secretname: "myrds-conn"
 ```
 
-(Optional) Define RDS component in an application
+Apply the application into the K8s system.
 
-<details>
+The database provision will take some time (> 5 min) to be ready.
 
-Let's first create an [Appfile](../developers/learn-appfile.md). We will claim an RDS instance with workload type of `rds`. You may need to change the variables of the `database` service to reflect your configuration.
+// TBD: add status check , or should database is created result.
 
-```bash
-$ cat << EOF > vela.yaml
-name: test-rds
-
-services:
-  database:
-    type: rds
-    name: alibabaRds
-    storage: 20
-
-  checkdb:
-    type: webservice
-    image: nginx
-    name: checkdb
-    env:
-      - name: PGDATABASE
-        value: postgres
-      - name: PGHOST
-        valueFrom:
-          secretKeyRef:
-            name: db-conn
-            key: endpoint
-      - name: PGUSER
-        valueFrom:
-          secretKeyRef:
-            name: db-conn
-            key: username
-      - name: PGPASSWORD
-        valueFrom:
-          secretKeyRef:
-            name: db-conn
-            key: password
-      - name: PGPORT
-        valueFrom:
-          secretKeyRef:
-            name: db-conn
-            key: port
-EOF
+
+## Consuming
+
+In this section, we will consume the cloud resources created.
+
+> ** Note: We highly recommend that you should split the cloud resource provision and consuming in different applications.**
+** Because the cloud resources can have standalone Lifecycle Management.**
+> But it also works if you combine the resources provision and consuming within an App.
+
+### Step 1 Define a ComponentDefinition consume from secrets
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  name: webserver
+  annotations:
+    definition.oam.dev/description: "webserver to consume cloud resources"
+spec:
+  workload:
+    definition:
+      apiVersion: apps/v1
+      kind: Deployment
+  schematic:
+    cue:
+      template: |
+        output: {
+        	apiVersion: "apps/v1"
+        	kind:       "Deployment"
+        	spec: {
+        		selector: matchLabels: {
+        			"app.oam.dev/component": context.name
+        		}
+        		template: {
+        			metadata: labels: {
+        				"app.oam.dev/component": context.name
+        			}
+        			spec: {
+        				containers: [{
+        					name:  context.name
+        					image: parameter.image
+
+        					if parameter["cmd"] != _|_ {
+        						command: parameter.cmd
+        					}
+        					env: [{
+        						name:  "DB_NAME"
+        						value: mySecret.dbName
+        					}, {
+        						name:  "DB_PASSWORD"
+        						value: mySecret.password
```

Comment:
IMHO, suggesting platform builders to use `valueFrom` is more secure than `value`. We should hesitate before exposing sensitive information like `password`, `accesskey` in spec.